File: eHour-persistence/src/main/java/net/rrm/ehour/persistence/backup/dao/RestoreDao.java
package net.rrm.ehour.persistence.backup.dao;

import net.rrm.ehour.domain.DomainObject;

import java.io.Serializable;

/**
 * @author thies (Thies Edeling - thies@te-con.nl)
 *         Created on: Nov 19, 2010 - 12:07:04 AM
 */
public interface RestoreDao
{
    <T extends DomainObject<?, ?>> Serializable persist(T object);

    <T, PK extends Serializable> T find(PK primaryKey, Class<T> type);

    void flush();

    <T> void delete(Class<T> type);
}

File: eHour-persistence/src/main/scala/net/rrm/ehour/persistence/backup/dao/RestoreDao.scala
package net.rrm.ehour.persistence.backup.dao

import java.io
import java.io.Serializable

import net.rrm.ehour.domain.DomainObject

trait RestoreDao {
  def persist[T <: DomainObject[_, _]](obj: T): io.Serializable

  def find[T, PK <: Serializable](primaryKey: PK, obj: Class[T]): T

  def flush()

  def delete[T](obj: Class[T])
}